app/javascript/admin/controllers/config_controller.js

```diff
@@ -246,4 +259,68 @@ export default class ConfigController extends Controller {
     }
   }
   // AUTH PROVIDERS FUNCTIONS END
+
+  enabledProvidersWithMissingKeys() {
+    const providersWithMissingKeys = [];
+    document
+      .querySelectorAll('[data-enable-auth="true"]')
+      .forEach((provider) => {
+        const providerName = provider.dataset.authProviderEnable;
+        const officialName = provider.dataset.authProviderOfficial;
+        if (
+          document.getElementById(`site_config_${providerName}_key`).value
+            .length === 0 ||
+          document.getElementById(`site_config_${providerName}_secret`).value
+            .length === 0
+        ) {
+          providersWithMissingKeys.push(officialName);
+        }
+      });
+
+    return providersWithMissingKeys;
+  }
+
+  generateProvidersList(providers) {
+    let list = '';
+    for (let i = 0; i < providers.length; i++) {
+      list += `<li>${providers[i]}</li>`;
+    }
+    return list;
+  }
```

Comment:
I think we can use `forEach` here (like you do [earlier in the file](https://github.com/forem/forem/pull/11750/files#diff-caedbb7d7984a39e9a441e5df7c79b425ec24ebe6d54b4e50261e0634c183793R267), which will make this easier to read :) 
https://alligator.io/js/foreach-vs-for-loops